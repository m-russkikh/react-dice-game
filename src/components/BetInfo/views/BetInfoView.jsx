import React from 'react';

import { BetInfoViewPropType } from '../propTypes';


function BetInfoView({
  betNumber, sign, chance, payout,
}) {
  return (
    <div>
      <div>number {sign} <b>{betNumber}</b></div>
      <div>chance: <b>{chance}%</b></div>
      <div>payout: <b>{payout}x</b></div>
    </div>
  );
}

BetInfoView.propTypes = BetInfoViewPropType;

export default BetInfoView;
