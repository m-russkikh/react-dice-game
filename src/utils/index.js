import { DEFAULT_BALANCE } from '../constants/balance';
import { MAX_BET_NUMBER } from '../constants/diceGame';

export function getPreloadedState() {
  let balance = {
    value: DEFAULT_BALANCE,
  };

  try {
    const serializedBalance = localStorage.getItem('balance');

    if (serializedBalance) {
      balance = JSON.parse(serializedBalance);
    }
  } catch (err) {
  //  ignore errors
  }

  return { balance };
}

export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getBetNumber() {
  return Math.floor(Math.random(0, 1) * MAX_BET_NUMBER) + 1;
}
