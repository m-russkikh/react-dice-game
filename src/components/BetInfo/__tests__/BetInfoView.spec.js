import { getShallowWrapper } from '../../../utils/testUtils';
import BetInfoView from '../views/BetInfoView';


describe('BetInfoView', () => {
  let props;
  let wrapper;

  beforeAll(() => {
    props = {
      betNumber: 20,
      sign: '<=',
      chance: 20,
      payout: 5,
    };
    wrapper = getShallowWrapper(BetInfoView, props);
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
