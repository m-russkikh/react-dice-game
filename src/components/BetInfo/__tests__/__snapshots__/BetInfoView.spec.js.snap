// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetInfoView render should display correct 1`] = `
"<div>
  <div>
    number 
    &lt;=
     
    <b>
      20
    </b>
  </div>
  <div>
    chance: 
    <b>
      20
      %
    </b>
  </div>
  <div>
    payout: 
    <b>
      5
      x
    </b>
  </div>
</div>"
`;
